workflow/metrics/labels.go

```diff
@@ -1,6 +1,15 @@
 package metrics
 
 const (
+	labelBuildVersion      string = `version`
+	labelBuildPlatform     string = `platform`
+	labelBuildGoVer        string = `goversion`
+	labelBuildDate         string = `build`
+	labelBuildCompiler     string = `compiler`
+	labelBuildGitCommit    string = `commit`
+	labelBuildGitTreeState string = `treestate`
+	labelBuildGitTag       string = `tag`
```

Comment:
```suggestion
	labelBuildGoVersion    string = `go_version`
	labelBuildDate         string = `build_date`
	labelBuildCompiler     string = `compiler`
	labelBuildGitCommit    string = `git_commit`
	labelBuildGitTreeState string = `git_tree_state`
	labelBuildGitTag       string = `git_tag`
```

same as above for docs, matching `argo version` and splitting words with underscores per OTel convention

Reply:
Done

Reply:
small follow-up: https://github.com/argoproj/argo-workflows/pull/13269#discussion_r1669061361